src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMappingProvider.cs

```diff
@@ -58,5 +59,41 @@ public void AddOrUpdatePackageSourceMappingSourceItem(PackageSourceMappingSource
 
             _settings.SaveToDisk();
         }
+
+        public void SavePackageSourceMappings(IReadOnlyList<PackageSourceMappingSourceItem> packageSourceMappingsSourceItems)
+        {
+            if (packageSourceMappingsSourceItems == null)
+            {
+                throw new ArgumentNullException(nameof(packageSourceMappingsSourceItems));
+            }
+
+
+            var existingSettingsLookup = GetPackageSourceMappingItems();
+
+            foreach (var sourceMappingItem in packageSourceMappingsSourceItems)
+            {
+                if (existingSettingsLookup != null)
```

Comment:
Could be place if condition outside the for loop? I believe this value doesn't change unless we to another lookup from settings file.

Reply:
Yes. Thanks for catching this